index.rst

```diff
@@ -48,14 +48,135 @@
+The camera will publish the observation ID as soon as the observation starts but any implementation involving a Butler would not be able to work unless an additional system was monitoring the observation ID from the camera and created a stub exposure record that would be fully populated later on.
+
+During observing there can be time gaps between exposures for various reasons.
+It could be helpful for those time gaps to be displayed as part of the exposure log user interface.
+Rather than storing annotations for time gaps (was it weather? was it a fault and a link to the Jira ticket?) in a completely distinct system, there are advantages to be able to store them along with exposure comments.
+Such gaps could coexist with real exposure records as virtual exposures (possible tagged with an ID related to the exposure that closed the gap) but the Butler dimension record system would likely not be amenable to such a concept.
+
+A further complication for the summit is how such annotations and flags can be added by remote observers and whether a view should be available at the USDF with minimal time lag.
+It is reasonable for an instrument scientist to want to make comments the morning after observing but where should those comments reside?
+If editing is allowed off the mountain should the summit instance receive those comments or should the summit only contain messages made on the summit instance and the source of truth be shifted to the data facility when the night is over?
+
+At a data facility all the output datasets already exist and virtual datasets (as opposed to virtual dimension records) are a concept that is fully supported by the Butler.
+The issue mainly becomes one of how to integrate dataset annotations and analysis bots into the campaign management system.
+For an integrated butler or butler adjacent implementation, transferring records between registries would be similar to that involved in transferring other records, although the butler adjacent approach would require some thought on how integrated those plugins are with registry.
+
+Open Questions
+==============
+
+* Should we allow commenting on any dimension record?
+  That would be the more general solution although it is only really helpful for ``visit`` and that would normally inherit the properties of ``exposure`` through provenance.
+* Should it be possible to constrain a graph build by either a tag or a quality flag?
+* Should dataset provenance affect the graph build if a parent dataset has a particular quality flag set even though the specific input dataset does not?
+* Summit users require the ability to tag specific error conditions such as vignetting or bad focus.
+  These are annotations against exposure records rather than explicit datasets and is somewhat orthogonal to the limited set of proposed quality flags.
+  Should Butler support tags on dimensions?
```

Comment:
I think there are a few things pushing us towards having something like this.  For example, DMTN-181 wants "data ID sets" to be not just uploadable from files, but persisted in the Registry DB for provenance.  I am not certain that's actually a useful kind of provenance (I plan to discuss this more in DMTN-220), but I think it's not hard for us to provide on top of other things that I do think we need to provide.  And "named, persistent data ID sets" are pretty much TAGGED collections for dimensions - I just wouldn't _actually_ make them TAGGED collections, because "data IDs can't be in collections" is still important conceptually.